extensions/spark/kyuubi-spark-lineage/src/main/scala/org/apache/kyuubi/plugin/lineage/helper/SparkSQLLineageParseHelper.scala

```diff
@@ -87,20 +119,26 @@ trait LineageParser {
   private def joinRelationColumnLineage(
       parent: AttributeMap[AttributeSet],
       relationAttrs: Seq[Attribute],
-      tableName: String = ""): AttributeMap[AttributeSet] = {
+      qualifier: Seq[String] = Seq()): AttributeMap[AttributeSet] = {
     val relationAttrSet = AttributeSet(relationAttrs)
     if (parent.nonEmpty) {
       parent.map { case (k, attrs) =>
         k -> AttributeSet(attrs.collect {
           case attr if relationAttrSet.contains(attr) =>
-            attr.withName(Seq(tableName, attr.name).filter(_.nonEmpty).mkString("."))
+            attr.withQualifier(qualifier)
+          case attr
+              if attr.qualifier.nonEmpty && attr.qualifier.last.equalsIgnoreCase(
+                SUBQUERY_COLUMN_IDENTIFIER) =>
+            attr.withQualifier(attr.qualifier.init)
+          case attr if attr.name.equalsIgnoreCase(AGGREGATE_COLUMN_IDENTIFIER) =>
+            attr.withQualifier(qualifier)
```

Comment:
can not we do `attr.withQualifier(qualifier)` for all cases ?

Reply:
Not all of them are suitable to use `withQualifier`, here it is more convenient and clear to use `withName`